### PR TITLE
API for decoration at caret and paragraph

### LINKED ISSUE #300: API for decoration at caret and paragraph
RTA needs to expose the current `Decoration` where the caret is, and the current `ParagraphDecoration` at the current paragraph.

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #300

### CODE REVIEW

File: rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java [hunk 1]
```diff
@@ -321,6 +322,28 @@ public final void setSkinTone(EmojiSkinTone value) {
     public final int getCaretPosition() { return caretPosition.get(); }
     public final ReadOnlyIntegerProperty caretPositionProperty() { return caretPosition.getReadOnlyProperty(); }
 
+
+    /**
+     * The current decoration at the caret.
+     */
+    final ReadOnlyObjectWrapper<Decoration> decorationAtCaretProperty = new ReadOnlyObjectWrapper<>(this, "decorationAtCaret");
```
Comment (abhinayagarwal) [inline rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java:329]: s/decorationAtCaretProperty/decorationAtCaret

File: rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java [hunk 2]
```diff
@@ -321,6 +322,28 @@ public final void setSkinTone(EmojiSkinTone value) {
...
+       return decorationAtCaretProperty.getReadOnlyProperty();
+    }
+    public final Decoration getDecorationAtCaret() {
+       return decorationAtCaretProperty.get();
+    }
+
+    /**
+     * The paragraph decoration at the current paragraph.
+     */
+    final ReadOnlyObjectWrapper<ParagraphDecoration> decorationAtParagraphProperty = new ReadOnlyObjectWrapper<>(this, "decorationAtParagraph");
```
Comment (abhinayagarwal) [inline rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java:340]: s/decorationAtParagraphProperty/decorationAtParagraph

**Review by abhinayagarwal — APPROVED** [review]
Overall, it looks good. A minor comment on renaming variables to make them smaller.

File: rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java (hunk 1 above)
Comment (jperedadnr) [reply rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java]: done

File: rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java (hunk 2 above)
Comment (jperedadnr) [reply rta/src/main/java/com/gluonhq/richtextarea/RichTextArea.java]: done

**Review by abhinayagarwal — APPROVED** [review]

**Pull request merged**